### PR TITLE
some op test

### PR DESCRIPTION
**Pull request by xiezipeng-ML**
- [x] oneflow.cat
- [x] oneflow.nn.Dropout
- [x] oneflow.new_ones
- [x] oneflow.Tensor.diag
- [x] oneflow.diag

### CODE REVIEW

File: python/oneflow/test/modules/test_cat.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    def test_cat_with_random_data(test_case):
+        device = random_device()
+        x = random_pytorch_tensor(
+            ndim=2, 
+            dim0=random(), 
+            dim1=random()
+            ).to(device)
+        return torch.cat(
+            (x, x, x),
+            random(0,1).to(int)
```
Comment (BBuf) [inline python/oneflow/test/modules/test_cat.py:40]: random(0, 2).to(int)吧

File: python/oneflow/test/modules/test_diag.py [hunk 2]
```diff
@@ -22,134 +22,34 @@
...
+        return torch.diag(x)
+
+    @autotest()
+    def test_diag_other_dim(test_case):
+        device = random_device()
+        x = random_pytorch_tensor(ndim=2, dim0=random(), dim1=random()).to(device)
+        return torch.diag(x)
+
+    @autotest()
+    def test_tensor_diag_one_dim(test_case):
```
Comment (BBuf) [inline python/oneflow/test/modules/test_diag.py:43]: 移动到test_tensor.py

File: python/oneflow/test/modules/test_new_ones.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+import unittest
+import oneflow as flow
+import oneflow.unittest
+from torch._C import dtype
+from automated_test_util import *
+
+
+@flow.unittest.skip_unless_1n1d()
+class Test_New_ones(flow.unittest.TestCase):
```
Comment (BBuf) [inline python/oneflow/test/modules/test_new_ones.py:22]: 这个没有必要了，我已经添加过了，在test_constant.py，可以直接把这个文件删掉

File: python/oneflow/test/modules/test_dropout.py [hunk 4]
```diff
@@ -22,91 +22,32 @@
...
-            _test_dropout_backward_p0,
-            _test_dropout_backward_p1,
-            _test_dropout_eval,
-            _test_dropout_with_generator,
-        ]
-        arg_dict["shape"] = [(2, 3), (2, 3, 4), (2, 3, 4, 5)]
-        arg_dict["device"] = ["cpu", "cuda"]
-        for arg in GenArgList(arg_dict):
-            arg[0](test_case, *arg[1:])
+class TestCat(flow.unittest.TestCase):
```
Comment (BBuf) [inline python/oneflow/test/modules/test_dropout.py:29]:
```suggestion
class TestDropout(flow.unittest.TestCase):
```

File: python/oneflow/test/modules/test_dropout.py [hunk 5]
```diff
@@ -22,91 +22,32 @@
...
+    @autotest()
+    def test_dropout(test_case):
+        device = random_device()
+        x = random_pytorch_tensor(ndim=random(),dim0=random()).to(device)
+        m = torch.nn.Dropout(p=0)
+        return m(x)
+
+    @autotest()
+    def test_dropout_p1(test_case):
```
Comment (BBuf) [inline python/oneflow/test/modules/test_dropout.py:38]: 测一下p!=0&&p!=1的随机值

File: python/oneflow/test/modules/test_cat.py (hunk 1 above)
Comment (xiezipeng-ML) [reply python/oneflow/test/modules/test_cat.py]: 已修改

File: python/oneflow/test/modules/test_diag.py (hunk 2 above)
Comment (xiezipeng-ML) [reply python/oneflow/test/modules/test_diag.py]: 已修改

File: python/oneflow/test/modules/test_dropout.py (hunk 5 above)
Comment (xiezipeng-ML) [reply python/oneflow/test/modules/test_dropout.py:38]: 这个dropout随机drop其tensor，所以没办法确保pytorch和oneflow的tensor被相同的drop，只能测试p=0和p=1的情况

**Review by BBuf — APPROVED** [review]
